pkg/cmd/factory/remote_resolver.go

```diff
@@ -59,21 +59,36 @@ func (rr *remoteResolver) Resolver() func() (context.Remotes, error) {
 		var hostname string
 		cachedRemotes = context.Remotes{}
 		sort.Sort(resolvedRemotes)
-		for _, r := range resolvedRemotes {
-			if hostname == "" {
-				if !knownHosts[r.RepoHost()] {
+
+		if ghinstance.Default() != ghinstance.OverridableDefault() {
```

Comment:
Could this `if` block return `cachedRemotes` early when done, so that instead of the large `else` block, the old logic that follows can remain intact? That way, we keep the diff minimal and keep closer to a style that the Go community seems to prefer (early returns)